### ISSUE TITLE
useFirstErrorOnly() breaks when errorFormatter is set.

### ISSUE THREAD

**Issue by d30jeff** [ctavan/express-validator#365]
I'm having a weird problem with the output of `getValidationResult().useFirstErrorOnly().mapped()` when `errorFormatter` option is passed.
 
There could be an error in my `errorFormatter` body though.

#### Input

```
{
  "email": "",
  "password": ""
}
```

#### Option

```
router.use(expressValidator({
  errorFormatter: function(param, msg, value) {
    return {
      field: param,
      errorMessage: msg,
      givenValue: value
    }
  }
}))
```

#### Validation

```
req.checkBody('password')
  .notEmpty().withMessage('required')
  .isLength(6,).withMessage('not_long_enough')

req.checkBody('email')
  .notEmpty().withMessage('required')
```

#### Get validation result

```
let errors = await req.getValidationResult()

if (errors.isEmpty() === false) {
  return res.status(400).json({
    type: 'validation',
    errors: errors.useFirstErrorOnly().mapped()
  })
}
```

#### Given output

```
{
    "type": "validation",
    "errors": {
        "undefined": {
            "field": "password",
            "errorMessage": "required",
            "givenValue": ""
        }
    }
}
```

#### Expected output

*Note: I'm getting this nicely formatted error when I removed `errorFormatter` in **options***, which is what I wanted.

```
{
    "type": "validation",
    "errors": {
        "password": {
            "param": "password",
            "msg": "required",
            "value": ""
        },
        "email": {
            "param": "email",
            "msg": "required",
            "value": ""
        }
    }
}
```

Thanks

**Comment by gustavohenke** [comment]
The `param` key is required in the returned object. At least for now...

**Comment by gustavohenke** [comment]
This *should* be fixed now with the v4.1.0 release.
Now the error formatter is applied *during* the validation errors retrieval.

_* In case you haven't yet, please read about [the v4.0.0 release](https://github.com/ctavan/express-validator/releases/tag/v4.0.0)._
